include/sqlpp11/compat/make_unique.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+/*
+ * Copyright (c) 2023, Vesselin Atanasov
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ *   Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ *   Redistributions in binary form must reproduce the above copyright notice, this
+ *   list of conditions and the following disclaimer in the documentation and/or
+ *   other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <memory>
+
+#ifdef _MSVC_LANG
+#define CXX_STD_VER _MSVC_LANG
+#else
+#define CXX_STD_VER __cplusplus
+#endif
+
+namespace sqlpp
+{
+  namespace compat
+  {
+    template <typename T, typename... Args>
+    inline std::unique_ptr<T> make_unique(Args&&... args)
```

Comment:
You should be able to use a function alias for C++14 and above, see for instance:

https://www.fluentcpp.com/2017/10/27/function-aliases-cpp/

Reply:
I also thought about using a templated variable, but unfortunately function aliases implemented via templated variables, don't support type deduction, so for `make_unique` we would have to specify the types for all the parameters which makes it more difficult to use compared to the original `std::make_unique`. Below is a sample program which shows the problem
```
#include <memory>

template <typename T, typename... Args>
const auto custom_make_unique = std::make_unique<T, Args...>;

int main()
{
	// Works fine with C++14 and newer
	[[maybe_unused]] auto var_1 = std::make_unique<std::pair<int, int>> (1, 2);

	// Works fine with C++14 and newer
	[[maybe_unused]] auto var_2 = custom_make_unique<std::pair<int, int>, int, int> (1, 2);

	// Fails with C++14 and newer, with "error: too many arguments to function"
	// because templated function aliases via templated variabled don't support type deduction
	[[maybe_unused]] auto var_3 = custom_make_unique<std::pair<int, int>> (1, 2);
	return 0;
}

```

Reply:
That's sad, but we'll have to live with it, I guess. Thanks for looking into it!